recipe-client-addon/lib/PreferenceExperiments.jsm

```diff
@@ -142,6 +142,16 @@ this.PreferenceExperiments = {
       previousPreferenceValue: Preferences.get(preferenceName, undefined),
     };
 
+    if (experiment.previousPreferenceValue) {
+      const previousType = typeof experiment.previousPreferenceValue;
+      const givenType = typeof preferenceValue;
```

Comment:
Also, the docs for `getPrefType` doesn't mention it, but the `PREF_BOOL` return values are constants defined on `Services.prefs`. So `Services.prefs.PREF_BOOL` can be used to check if the preference is a boolean, and so forth.